### PR TITLE
:running: Update PR template to match CAPI

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
This PR updates the PR template to match what we're doing in Cluster API.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,25 +1,16 @@
-<!--  Thanks for sending a pull request!  Here are some tips for you:
-1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/guide/pull-requests.md#the-pull-request-submit-process and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
-2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
-3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
-4. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-5. If this PR changes image versions, please title this PR "Bump <image name> from x.x.x to y.y.y."
--->
+<!-- please add a icon to the title of this PR (see VERSIONING.md), and delete this line and similar ones -->
```
Comment (ncdc) [inline .github/PULL_REQUEST_TEMPLATE.md]: We probably need a VERSIONING.md, both in CAPI & here?

Comment (vincepri) [reply .github/PULL_REQUEST_TEMPLATE.md]: Updated and linked to CAPI's one

**Pull request merged**